src/customers/dto/update-customer.dto.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  IsEmail,
+  IsNotEmpty,
+  IsOptional,
+  IsString,
+  IsStrongPassword,
+  IsUUID,
+} from 'class-validator';
+
+import { ApiProperty } from '@nestjs/swagger';
+
+export class UpdateCustomerDto {
```

Comment:
**praise:** nice job separating dto per need (e.g.: update, create and return dto). I like doing this way to avoid common mistakes in which you could allow the client modify properties you don't want to. This is basically a command pattern, but note it could possibly grow out of control if you have a lot of different post/put operations and some people might be against this pattern (of course is a subjective reasoning)